src/qz/printer/status/printer/CupsPrinterStatusMap.java

```diff
@@ -918,4 +921,15 @@ public NativePrinterStatus getParent() {
     public Object getRawCode() {
         return name().toLowerCase(Locale.ENGLISH).replace("_", "-");
     }
+
+    // Bugfix for issue #1017
+    // Removes redundant '-warning' '-report' from SNMP originated statuses, with some exceptions
+    private static String statusFix(String cupsString) {
+        ArrayList exceptions = new ArrayList(Arrays.asList("offline-report", "cups-insecure-filter-warning", "cups-missing-filter-warning"));
```

Comment:
Please convert exceptions from the enums.

Reply:
... arguably, you could programatically create this list too.  I'm going to take a swing at a small refactor and send what I come up with.